src/Workspaces/CoreTest/WorkspaceTests/CommandLineProjectTests.cs

```diff
@@ -110,5 +111,32 @@ public void TestAdditionalFiles()
             Assert.Equal("foo.cs", firstDoc.Name);
             Assert.Equal("bar.cs", secondDoc.Name);
         }
+
+        [Fact]
+        public void TestAnalyzerReferences()
+        {        
+            var pathToAssembly = typeof(object).Assembly.Location;
+            var assemblyBaseDir = PathUtilities.GetDirectoryName(pathToAssembly);
+            var assemblyName = PathUtilities.GetRelativePath(assemblyBaseDir, pathToAssembly);
+            string assemblyBaseDirName;            
+            if (PathUtilities.GetPathRoot(assemblyBaseDir) != assemblyBaseDir)
+            {
+                var assemblyBaseBaseDir = PathUtilities.GetDirectoryName(assemblyBaseDir);
+                assemblyBaseDirName = PathUtilities.GetRelativePath(assemblyBaseBaseDir, assemblyBaseDir);
+            }
+            else // e.g. pathToAssembly = C:\mscorlib.dll.
+            {
+                assemblyBaseDirName = "..";
+            }
+            var relativePath = PathUtilities.CombinePossiblyRelativeAndRelativePaths(
+                PathUtilities.CombinePossiblyRelativeAndRelativePaths("..", assemblyBaseDirName), assemblyName);
```

Comment:
Is it necessary for this test to actually create a path that has a prefix like this? Is:

    var pathToAssembly = typeof(object).Assembly.Location;
    var assemblyBaseDir = Path.GetDirectoryName(pathToAssembly);
    string commandLine = @"foo.cs /a:" + Path.GetFileName(pathToAssembly);

not sufficient?

It just seems like a lot of non-trivial setup that is really easy to get lost in really quickly.